### DISCUSSION TITLE
Returning a List of Images in Appearance Order

### DISCUSSION THREAD

**Discussion by WilliamHerbert** [fastily/jwiki#32] [Q&A] [answered]
I'm working on something and want to blindly pull the first image from an article, assuming that it is most likely an actor/celebrity/politician's photo.  When I call getImagesOnPage, the result is a list of images in alphabetical order.

I was looking through the code to see where the sorting was taking place, and I wasn't quite able to figure it out.  Is the sort already applied to the JSON/GSON object that is being parsed?  I'm just trying to figure out if there's a way I can squeeze in an option for on page appearance order, either by forking and just doing my own thing, or forking and bringing it back as a feature.

Any hints would be appreciated!

**Comment by fastily** [comment]
Hi @WilliamHerbert,

As far as I know, this is not possible with the API.  You may have better luck scraping and parsing the HTML of the page(s) in question

**Comment by NicolasWinsten** [comment, marked as answer]
If you want the featured image, you'd have a much easier time scraping it off of the "Information" page: https://en.wikipedia.org/w/index.php?title=Rowan_Atkinson&action=info

**Reply by WilliamHerbert** [reply]
Thanks for the pointer! I didn't even know this was a thing.
